R/createCompDbPackage.R

```diff
@@ -727,10 +727,11 @@ createCompDb <- function(x, metadata, msms_spectra, path = ".") {
     cols <- cols[!cols %in% c("spectrum_id", "compound_id", "mz", "intensity")]
     cols <- cols[!cols %in% colnames(x)]
     n <- nrow(x)
-    for (col in cols) {
-        cn <- colnames(x)
-        x <- cbind(x, rep(as(NA, .required_msms_spectrum_columns[col]), n))
-        colnames(x) <- c(cn, col)
+    if (length(cols)) {
+        x_new <- as.data.frame(
+            lapply(.required_msms_spectrum_columns[cols],
```

Comment:
@jianhong I don't see a way to vectorize that functionality in any other way, as I'm creating here a `data.frame` with columns containing `NA` values of a certain data type (depending on the missing columns).